### ISSUE TITLE
Fix documentation for F1 metric: average=None

### ISSUE THREAD

**Issue by Hrant-Khachatrian** [PyTorchLightning/pytorch-lightning#5236]
## 📚 Documentation

The [documentation](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/pytorch_lightning/metrics/classification/f_beta.py#L59) for `metrics.classification.F1` and `FBeta` says `'none'` is the value for `average` argument, but the code [expects](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/pytorch_lightning/metrics/classification/f_beta.py#L102) `None`.

**Comment by awaelchli** [comment]
are you interested in sending a PR with the update?

**Comment by Hrant-Khachatrian** [comment]
@awaelchli not sure which is the correct version :) The code is the "truth"? and the docs should be updated to match it?

**Comment by awaelchli** [comment]
probably None is what a python user expects? I'm not sure what the original intention was, maybe it was inspired by sklearn @SkafteNicki?

**Comment by awaelchli** [comment]
Here is the F1 score and they document it as None (not "none")
https://scikit-learn.org/stable/modules/generated/sklearn.metrics.f1_score.html

**Comment by Hrant-Khachatrian** [comment]
`None` is more intuitive for me, including because of the sklearn, as you mentioned. But then there are similar parameters in PyTorch that use `'none'`, e.g. `reduction` for [`NLLLoss`](https://pytorch.org/docs/stable/generated/torch.nn.NLLLoss.html). 

There is `dist_reduce_fx` somewhere in `Metric` class of Lightning, which uses `'cat'` for a similar purpose... Although one might argue that this is not too closely related.

Then there is [`Recall`](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/pytorch_lightning/metrics/classification/precision_recall.py#L115), which does not have `None` option at all...

So this looks complicated. I'm not sure I know all related functions in Lightning to check for consistency.

**Comment by SkafteNicki** [comment]
@awaelchli So the story is that prior to the v1.0 refactor of metrics we used `'none'` as we were using PyTorch as the initial inspiration. However, as time passed we have begun to lean more towards sklearns interface (`None`) because it is well established framework for doing metric computations. So to both be backwards compatible but also support people used to sklearn, the idea was to simply allow both options. This can be seen in the function that actually does the reduction:
https://github.com/PyTorchLightning/pytorch-lightning/blob/c479351a938240fbda6774a404494ee399ff361a/pytorch_lightning/metrics/utils.py#L253-L274
Therefore, @Hrant-Khachatrian if you could make sure that everywhere supports **both** `'none'` and `None` as an option that would be great :]

**Comment by SkafteNicki** [comment]
Fixed by PR #5880. Closing.
